client/src/util/helpers.ts

```diff
@@ -305,4 +306,67 @@ export default {
       return translation + " " + translationSuffix;
     }
   },
+
+  /**
+   * Detects whether a given DOM element is visible on screen.
+   *
+   * Note: for older browsers that do not support IntersectionObserver, this
+   * hook will always return FALSE by default.
+   *
+   * Borrowed from https://stackoverflow.com/questions/45514676/react-check-if-element-is-visible-in-dom
+   */
+  useOnScreen(ref: React.RefObject<any>) {
```

Comment:
should this be `isOnScreen` since it returns a boolean? wait, does it return a boolean?

Reply:
oh i think it might be a [React hook](https://reactjs.org/docs/hooks-intro.html), those always start with `use` by convention...

Reply:
yes ditto what @toolness said, I remember being a bit confused by that too